leaf-ui/rappid-extensions/BackboneModelsIntention.js

```diff
@@ -39,17 +38,526 @@ var EvolvingFunctionBBM = Backbone.Model.extend({
+
+        repCount = stopIndex - startIndex;
+
+        // Original Function
+        // this.removeRepFuncSegments();
+        // // find the index of the FuncSegment with start time time 1
+        // var startIndex = 0;
+        // while (this.functionSegList[startIndex].funcStart !== time1) {
+        //     startIndex++;
+        // }
+
+        // var repFuncSegments = [];
+
+        // // push and remove, until we see a segment with our desired FuncEnd time
+        // while (this.functionSegList[startIndex].funcStop !== time2) {
+        //     repFuncSegments.push(this.functionSegList[startIndex]);
+        //     this.functionSegList.splice(startIndex, 1);
+        // }
+
+        // // push and remove the last segment
+        // repFuncSegments.push(this.functionSegList[startIndex]);
+        // this.functionSegList.splice(startIndex, 1);
+
+
+        // // create and add a new RepFuncSegment
+        // var repFuncSegment = new RepFuncSegment(repFuncSegments);
+        // this.functionSegList.splice(startIndex, 0, repFuncSegment);        
+    },            
+
+    /**
+     * If a RepFuncSegment exists in this EvolvingFunction's
+     * functionSegList, retrieve the FuncSegments in the RepFuncSegment,
+     * remove the RepFuncSEgment and add the retrieved FuncSegments back
+     * into their correct positions in functionSegList
+     * 
+     * Updated: All of the parameters for a repeating segment are reset to the default,
+     * which means there is no loner a repeating function segment in the EvolvingFunctionBBM     *  
+     *
+     * If a RepFuncSegment does not exist in functionSegList, this function
+     * does nothing
+     */
+    removeRepFuncSegments: function() {
+        hasRepeat = false;
+        repStart = null;
+        repStop = null;          
+        repCount = null;         
+        repAbsTime = null;  
+        // Original Function
+        // var repIndex = this.getRepFuncSegmentIndex();
+        // if (repIndex === -1) {
+        //     return;
+        // }
+
+        // var repFuncSegment = this.functionSegList[repIndex];
+        // // remove RepFuncSegment object from array
+        // this.functionSegList.splice(repIndex, 1);
+
+        // // add the FuncSegments back into the array
+        // var j = repIndex;
+        // for (var i = 0; i < repFuncSegment.functionSegList.length; i++) {
+        //     this.functionSegList.splice(j, 0, repFuncSegment.functionSegList[i]);
+        //     j++;
+        // }
+    },    
+    
+    /**
+     * Returns the epoch boundary where the repeat segment
+     * starts
+     *
+     * Precondition: This EvolvingFunction must contain a RepFuncSegment
+     *
+     * @returns {String}
+     *   ex: 'A'
+     */
+    // TODO - fix this function 
+    getStartRepeatEpoch: function() {
+        if (this.repStart != 0) {
+            var repStartIndex = this.repStart.charCodeAt() - 64; // this should be the index of the first repeating segment
```

Comment:
Can we get a comment explaining why this is the index of the first repeating segment

Reply:
> Can we get a comment explaining why this is the index of the first repeating segment

I think the comment was changed to "index of the last repeating segment".

Reply:
> Can we get a comment explaining why this is the index of the first repeating segment

I think we have to discuss this further to determine if the functions actually work before adding in the comments you requested.